code/wdb/writedown.q

```diff
@@ -0,0 +1,22 @@
+/-endofperiod function to savedown by integer
+endofperiod:{[currp;nextp;data]
+	.lg.o[`endofperiod;"Received endofperiod. currentperiod, nextperiod and data are ",(string currp),", ", (string nextp),", ", .Q.s1 data]
+        /-check if there is more than 1 wdb processes
+	h:exec w from .servers.SERVERS where proctype=`wdb,not w=0N;
+        /-create a dictionary of process names and their start times
+	times:raze{@[;".proc.starttimeUTC";()][x]} each h;
+	if[.proc.starttimeUTC >max times;
+		@[`.;`upd;:;insert];
+		.wdb.currentpartition:`long$nextp;
+		:()];
+	/-If we are the new process, exit function, do not want to writedown
```

Comment:
Can we move this comment to above the if statement please. Also a note on why we need to update upd would be helpful :) "Setting variables so wdb can take over/become 'active' for this new period" something like this?